src/SSMSPlusSearch/UI/SchemaSearchControl.xaml.cs

```diff
@@ -46,5 +52,53 @@ public void Dispose()
             ViewModel?.Free();
             this.DataContext = null;
         }
+
+        #region Grid Events
+
+        /// <summary>
+        /// Initializes the current sort column and direction.
+        /// </summary>
+        /// <param name="sender">The products data grid.</param>
+        /// <param name="e">Ignored.</param>
+        private void objectsGrid_Loaded(object sender, RoutedEventArgs e)
+        {
+            DataGrid dataGrid = (DataGrid)sender;
+
+            // The current sorted column must be specified in XAML.
+            currentSortColumn = dataGrid.Columns.Where(c => c.SortDirection.HasValue).Single();
+            currentSortDirection = currentSortColumn.SortDirection.Value;
+        }
+
+        /// <summary>
+        /// Sets the sort direction for the current sorted column since the sort direction
+        /// is lost when the DataGrid's ItemsSource property is updated.
+        /// </summary>
+        /// <param name="sender">The parts data grid.</param>
+        /// <param name="e">Ignored.</param>
+        private void objectsGrid_TargetUpdated(object sender, DataTransferEventArgs e)
+        {
+            if (currentSortColumn != null)
+            {
+                currentSortColumn.SortDirection = currentSortDirection;
+            }
+        }
+
+        /// <summary>
+        /// Custom sort the datagrid since the actual records are stored in the
+        /// server, not in the items collection of the datagrid.
+        /// </summary>
+        /// <param name="sender">The parts data grid.</param>
+        /// <param name="e">Contains the column to be sorted.</param>
+        private void objectsGrid_Sorting(object sender, DataGridSortingEventArgs e)
+        {
+            //Save current sort column and sort direction
+            currentSortColumn = e.Column;
+            currentSortDirection = (e.Column.SortDirection != ListSortDirection.Ascending) ? ListSortDirection.Ascending : ListSortDirection.Descending;
+
+            //Get Current sort column name. e.g. SearchResult.Name => Name
+            ViewModel.CurrentSortColumn = currentSortColumn.SortMemberPath.Split('.')[1];
+            ViewModel.CurrentSortDirection = currentSortDirection;
+        } 
+        #endregion

```

Comment:
this block induces a lot of implicit knowledge:
SortMemberPath.Split('.')[1] assumes a particular path structure
dataGrid.Columns.Where(c => c.SortDirection.HasValue).Single() assumes some implicit setup in the xaml

What about multicolums sorting ?